### PR TITLE
User settings and logout

### PR DESCRIPTION
**Pull request by EmileCalixte**
| Q             | A
| ------------- | ---
| Is bugfix?    | No
| New feature?  | Yes
| Fixed issues  |

Logout button and user email & password settings

### CODE REVIEW

**Comment by EmileCalixte** [comment]
Review required !

File: frontend/src/components/pages/Settings/Settings.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            if(li.getAttribute('data-desktop-nav') === name) {
+                li.classList.add('active');
+                continue;
+            }
+
+            li.classList.remove('active');
+        }
+    }
+
+    componentDidMount() {
```
Comment (alexisljn) [inline frontend/src/components/pages/Settings/Settings.js:82]: Can you place this method just under constructor() ? I like to have sort of chronology in classes's methods order. I start by constructor, then the componentDidMount() lifecycle. Others methods are following, before I conclude with the render() lifecycle acting like the "return of the class".

File: frontend/src/components/pages/Settings/views/EmailSettings.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        const state = this.state;
+        state.formErrors = errors;
+        this.setState(state);
+
+        for(const field in this.state.formErrors) {
+            toastr.error(this.state.formErrors[field]);
+        }
+    }
+
+    componentDidMount = async () => {
```
Comment (alexisljn) [inline frontend/src/components/pages/Settings/views/EmailSettings.js:110]: Can you place this method just under constructor() ? I like to have sort of chronology in classes's methods order. I start by constructor, then the componentDidMount() lifecycle. Others methods are following, before I conclude with the render() lifecycle acting like the "return of the class".

**Review by alexisljn — COMMENTED** [review]
Strong and solid work ! I made few suggestions without requesting changes. Consider them and we'll talk about that

File: backend/controllers/SettingsController.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                    throw e;
+            }
+        }
+
+        this.response = {
+            email: user.email,
+        };
+    }
+
+    actionEditPassword = async (requestBody, accessTokenPayload) => {
```
[the remark this answers is not in the document]
Comment (EmileCalixte) [reply backend/controllers/SettingsController.js:114]: Yes, we could do that.

File: backend/controllers/SettingsController.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            this.statusCode = 422;
+            this.response = {
+                errors: {
+                    newEmail: "Cette adresse email est déjà utilisée",
+                },
+            };
+            return;
+        }
+
+        // await user.setEmail(requestBody.newEmail);
```
[the remark this answers is not in the document]
Comment (EmileCalixte) [reply backend/controllers/SettingsController.js:90]: Of course not ! 😄

File: frontend/src/components/pages/Settings/Settings.js (hunk 1 above)
Comment (EmileCalixte) [reply frontend/src/components/pages/Settings/Settings.js:82]: You're right

**Review by alexisljn — APPROVED** [review]
Nice !

**Pull request merged**